### PR TITLE
test client rpc

### PR DESCRIPTION
**Pull request by winsvega**
work in progress test client rpc
connects to https://github.com/ethereum/webthree-umbrella/issues/260

### CODE REVIEW

File: libwebthree/WebThree.cpp [hunk 1]
```diff
@@ -61,7 +62,9 @@ WebThreeDirect::WebThreeDirect(
 		if (_params.sealEngineName == "Ethash")
```
Comment (chriseth) [inline libwebthree/WebThree.cpp:63]: I think it would be better to use an enum as constructor argument for webthree and depending on that create the respective client.

File: eth/main.cpp [hunk 2]
```diff
@@ -1236,21 +1237,21 @@ int main(int argc, char** argv)
 					rpc::EthFace, rpc::DBFace, rpc::WhisperFace,
 					rpc::NetFace, rpc::Web3Face, rpc::PersonalFace,
 					rpc::AdminEthFace, rpc::AdminNetFace, rpc::AdminUtilsFace,
-					rpc::DebugFace
+					rpc::DebugFace, rpc::TestFace
```
Comment (chriseth) [inline eth/main.cpp]: The test interface should only be available if `eth` is started with `--test`.

This `--test` option should perform the following:
- add test rpc interface
- use NoProof engine
- accept blocks in the future

in effect, it should be possible to mine more than one block per second.

File: libweb3jsonrpc/Test.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file Bzz.h
```
Comment (chriseth) [inline libweb3jsonrpc/Test.h]: `Test.h`

File: libweb3jsonrpc/Test.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Test::Test(eth::Client& _eth): m_eth(_eth) {}
+
+bool Test::test_mineBlocks(int _number, int _timestamp)
+{
+	asClientTest(&m_eth)->mineBlocks(_number, _timestamp);
+	return true;
+}
+
+bool Test::test_addBlock(std::string const& _rlp)
```
Comment (chriseth) [inline libweb3jsonrpc/Test.cpp:38]: I think it would be better to have `addBlocks(unsigned _number)` (i.e. adds `_number` empty blocks) and unrelated to that `setTimestamp(unsigned _timestamp)` which sets the timestamp for the pending block. I.e. if I first call `setTimestamp` and then use `eth_call` and request the timestamp in the code, it returns that timestamp.

If the timestamp part is too complicated, do it in its own PR.

**Comment by chriseth** [comment]
Please rebase and supply a `nullptr` for the test interface if test is not enabled.

File: libweb3jsonrpc/Test.cpp (hunk 4 above)
Comment (winsvega) [inline libweb3jsonrpc/Test.cpp:38]: that is being done in a function above. 
Test::test_mineBlocks(int _number, int _timestamp)

it mines _number of empty blocks with _timestamp difference from the latest imported block

**Comment by winsvega** [comment]
done

File: libweb3jsonrpc/Test.cpp (hunk 4 above)
Comment (chriseth) [inline libweb3jsonrpc/Test.cpp:38]: Indeed, but I think it would be better to have two different functions for that functionality.

File: libwebthree/WebThree.cpp [hunk 5]
```diff
@@ -65,7 +63,9 @@ WebThreeDirect::WebThreeDirect(
...
 			m_ethereum.reset(new eth::EthashClient(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we));
 		else
-			m_ethereum.reset(new eth::Client(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we, _l));
+		if(_params.sealEngineName == "NoProof" && _testing)
+			m_ethereum.reset(new eth::ClientTest(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we));
+		else m_ethereum.reset(new eth::Client(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we, _l));
```
Comment (chriseth) [inline libwebthree/WebThree.cpp:68]: Newline after `else`

File: libwebthree/WebThree.cpp [hunk 6]
```diff
@@ -65,7 +63,9 @@ WebThreeDirect::WebThreeDirect(
 		if (_params.sealEngineName == "Ethash")
 			m_ethereum.reset(new eth::EthashClient(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we));
 		else
-			m_ethereum.reset(new eth::Client(_params, (int)_params.u256Param("networkID"), &m_net, shared_ptr<GasPricer>(), _dbPath, _we, _l));
+		if(_params.sealEngineName == "NoProof" && _testing)
```
Comment (chriseth) [inline libwebthree/WebThree.cpp:66]: `if` should move to previous line and have a space after it: `else if (`
